File: Converge/Models/CalendarEvent.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

using Microsoft.Graph;
using System.Collections.Generic;
using System.Linq;

namespace Converge.Models
{
    public class CalendarEvent
    {
        public CalendarEvent(Event e)
        {
            if (e.Subject.ToLower() == "focus time")
            {
                IsFocusTime = true;
            }
            IsAllDay = e.IsAllDay;
            Start = e.Start;
            End = e.End;
            Attendees = e.Attendees;
            Location = e.Locations.ToList().Find(l => l.LocationType == LocationType.ConferenceRoom);
            Id= e.Id;
        }

        public bool? IsAllDay { get; set; }

        public bool IsFocusTime { get; set; }

        public DateTimeTimeZone Start { get; set; }

        public DateTimeTimeZone End { get; set; }

        public IEnumerable<Attendee> Attendees { get; set; }

        public Location Location { get; set; }

        public string Id { get; set; }

        public bool IsWorkspaceBooking { get; set; }

        public string BuildingName { get; set; }
    }
}

File: Converge/ClientApp/src/types/CalendarEvent.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import * as MicrosoftGraph from "@microsoft/microsoft-graph-types";

interface CalendarEvent {
  isAllDay?: boolean;
  isFocusTime: boolean;
  start: MicrosoftGraph.DateTimeTimeZone;
  end: MicrosoftGraph.DateTimeTimeZone;
  attendees: MicrosoftGraph.Attendee[];
  isWorkspaceBooking?: boolean;
  location: MicrosoftGraph.Location;
  buildingName?: string;
  id: string;
}

export default CalendarEvent;
